File: java/matrix/src/main/java/Matrix.java
import java.util.*;

public class Matrix {

	private int[][] matrix;

    public Matrix(String matrixAsString) {
        String[] strRows = matrixAsString.split("\n");
        matrix = new int[strRows.length][];
        for(int i = 0; i < strRows.length; i++) {
        	matrix[i] = Arrays.stream(strRows[i].split("\\s")).mapToInt(num -> Integer.parseInt(num)).toArray();
        }
    }

    public int[] getRow(int rowNumber) {
    	return matrix[rowNumber];
    }

    public int[] getColumn(int columnNumber) {
    	int[] array = new int[matrix.length];
    	int i = 0;
    	for(int[] row : matrix) {
    		array[i] = row[columnNumber];
    		i++;
    	}
    	return array;
    }
}

File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self.matrix = []
        rows = matrix_string.splitlines()
        for row in rows:
        	self.matrix.append([int(num) for num in row.split()])

    def row(self, index):
        return self.matrix[index - 1]

    def column(self, index):
        return [row[index - 1] for row in self.matrix]
